platforms/jvm/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -1234,6 +1261,19 @@ public FileTree getCandidateClassFiles() {
         return getTestClassesDirs().getAsFileTree().matching(patternSet);
     }
 
+    /**
+     * Returns directories to scan for non-class-based test definition files.
+     *
+     * @return The candidate directories holding non-class-based test definition files.
+     * @since 9.3.0
+     */
+    @Incubating
+    @InputFiles
+    @SkipWhenEmpty
```

Comment:
I'm not sure what happens when a task has two properties with `SkipWhenEmpty`.  Do we check both of them and only skip if both are empty?